### PR TITLE
lazy property fetching via session.fetch()

### PR DESCRIPTION
**Pull request by gavinking**
So this here's my implementation of #204.

WDYT?

### CODE REVIEW

File: example/src/main/java/org/hibernate/example/reactive/Main.java [hunk 1]
```diff
@@ -46,7 +46,11 @@ public static void main(String[] args) {
 				//retrieve a Book
 				session -> session.find(Book.class, book1.id)
 						//print its title
-						.thenAccept( book -> out.println(book.title + " is a great book!") )
+						.thenCompose( book -> {
+							out.println(book.title + " is a great book!");
+							return session.fetch( book, Book_.isbn )
+									.thenAccept( isbn -> out.println(isbn + " is the ISBN") );
+						} )
```
Comment (gavinking) [inline example/src/main/java/org/hibernate/example/reactive/Main.java:53]: Folks, this is what it looks like to use this feature.

Anybody have any thoughts? Is this a good feature for us to have?

File: example/build.gradle [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+    }
+}
+
+apply plugin: 'org.hibernate.orm'
+
+hibernate {
+    enhance {
+        enableLazyInitialization= true
+        enableDirtyTracking = true
+        enableAssociationManagement = true
```
Comment (DavideD) [inline example/build.gradle:18]: Do you need all these 3 properties enabled?

Comment (gavinking) [reply example/build.gradle:18]: *That* I am not sure of. I *presume* we don't need `enableDirtyTracking`, but if we're going to enable the bytecode processor we might as well? It help, I presume?

And now that I realize what `enableAssociationManagement` is, I think I *don't* want it.

**Comment by gavinking** [comment]
Alright, I've no added a test for this stuff. I'm inclined to merge it.

File: example/src/main/java/org/hibernate/example/reactive/Main.java (hunk 1 above)
Comment (DavideD) [reply example/src/main/java/org/hibernate/example/reactive/Main.java:53]: LGTM

File: example/build.gradle [hunk 3]
```diff
@@ -18,3 +18,25 @@ dependencies {
...
+        classpath "org.hibernate:hibernate-gradle-plugin:${hibernateOrmVersion}"
+    }
+}
+
+apply plugin: 'org.hibernate.orm'
+
+hibernate {
+    enhance {
+        enableLazyInitialization = true
+        enableDirtyTracking = true
```
Comment (DavideD) [inline example/build.gradle:39]: If I disable dirtyTracking (`enableDirtyTracking=false`) the example fails when I run it. is that expected?

Comment (gavinking) [reply example/build.gradle:39]: It's working for me with:

    enableDirtyTracking = false

So, I dunno...

Comment (gavinking) [reply example/build.gradle:39]: I'm going to merge it @DavideD, I doubt that weird behavior of the bytecode enhancer is related to any bug in my code.

**Pull request merged**